### PR TITLE
chore: workflow documentation cleanup

### PR DESCRIPTION
**Pull request by caseyisonit**
We are starting to annotate our GitHub workflows and tasks so they are self-documented and easier to understand what is happening in them. 

## Motivation and context

- These files are not touched often and are pretty complex in their setup and abstractions. 
- Since we are open source, our repo should be heavily self-documented for the ease of contributors and ourselves
- update our linter to be current

## How has this been tested?

-   [ ] _Tests pass as expected_
    1. This PR passes all expected tests (CircleCi still is flaky)
    2. CircleCi Lint is passing

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: .prettierrc.yaml
Comment (caseyisonit) [inline .prettierrc.yaml:1]: Not sure why this file got picked up but nothing changed. I was playing with the tabWidth but reset it to 4 until we reach a decision as a team

File: projects/documentation/src/components/layout.ts
Comment (caseyisonit) [inline projects/documentation/src/components/layout.ts:1]: circleCi Lint fixes

File: projects/documentation/src/components/side-nav.ts
Comment (caseyisonit) [inline projects/documentation/src/components/side-nav.ts:1]: circleCi Lint fixes

File: projects/documentation/src/utils/templates.ts
Comment (caseyisonit) [inline projects/documentation/src/utils/templates.ts:1]: circleCi Lint fixes

File: tasks/build-lighthouse-comment.js [hunk 1]
```diff
@@ -159,14 +172,16 @@ export const buildLighthouseComment = (links, manifest, assertionResults) => {
...
+
+    // Compare resource usage between branch, main, and latest
     const { branchResources, mainResources, latestResources } =
         compareResourceUsage(
             branchResourcesRaw,
             mainResourcesRaw,
             latestResourcesRaw
         );
 
-    /* eslint-disable prettier/prettier */
```
Comment (caseyisonit) [inline tasks/build-lighthouse-comment.js:169]: this was unnecessary

File: tasks/build-preview-urls-comment.js [hunk 2]
```diff
@@ -12,74 +12,63 @@ governing permissions and limitations under the License.
...
-        ) {
-            acc.push(name);
-        }
-        return acc;
-    }, []);
-    return packageList;
-};
-
-const getHash = (context) => {
+const createHash = (context) => {
```
Comment (caseyisonit) [inline tasks/build-preview-urls-comment.js:18]: renamed to be semantically correct at what the function is doing

**Review by blunteshwar — COMMENTED** [review]
LGTM!
Once all the circle ci tests are passing, I think we are good to go!

**Comment by caseyisonit** [comment]
@blunteshwar cleaned up and passing!

File: .github/actions/setup-job/action.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    steps:
+        - name: Setup Node 20
+          uses: actions/setup-node@v4
+          with:
+              node-version: '20'
+              cache: 'yarn'
+              registry-url: 'https://registry.npmjs.org'
+
+        - name: Install dependencies
+          shell: bash
```
Comment (Rajdeepc) [inline .github/actions/setup-job/action.yml:14]: This is redundant!

**Review by Rajdeepc — CHANGES_REQUESTED** [review]
Good Start! Thanks for cleaning this up! One thing I would like to do is to make sure the existing workflow is working fine and do a sanity test!
1. Update a storybook and surface up changed VRTs
2. Make sure the proper has is getting generated for newly changed commits

File: .github/actions/setup-job/action.yml (hunk 3 above)
Comment (caseyisonit) [reply .github/actions/setup-job/action.yml:14]: @Rajdeepc which is redundant? the registry URL

**Review by Rajdeepc — APPROVED** [review]
Thanks for doing this! Great addition

**Pull request merged**